x-pack/plugin/rank-text-similarity/src/main/java/org/elasticsearch/xpack/rank/textsimilarity/TextSimilarityRankPlugin.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.rank.textsimilarity;
+
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.license.License;
+import org.elasticsearch.license.LicensedFeature;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.plugins.SearchPlugin;
+import org.elasticsearch.search.rank.RankBuilder;
+import org.elasticsearch.xcontent.NamedXContentRegistry;
+import org.elasticsearch.xcontent.ParseField;
+
+import java.util.List;
+
+public class TextSimilarityRankPlugin extends Plugin implements SearchPlugin {
```

Comment:
I mirrored the RRF plugin, I wasn't aware of any strong arguments for not having separate plugins.
Is it OK if I file a tech debt ticket to move these into `inference` along with RRF, rather than addressing it here?